Improved installation instructions for OpenShift

makuhama opened the pull request:
Instructions for creating the pull secret have been fixed.

Removed hard-wired namespace 'mongodb' in resources defined in enterprise-mongodb-openshift.yaml

Instead of instructing the reader to edit enterprise-mongodb-openshift.yaml the service accounts defined there are now linked to a secret named 'openshift-pull-secret'. This in sync with the instructions in 'openshift-install.md' and significantly reduces the amount of OpenShiftf specific instructions.

BenElgar commented:
Thank you for your contribution! We're reviewing this internally and tracking this as [CLOUDP-55732](https://jira.mongodb.org/browse/CLOUDP-55732) (private link).

makuhama commented:
Thanks for taking care of my PR. Just in case it matters: I have no access at all to the above mentioned JIRA ticket (not even to the project CLOUDP). Just signed up to MongoDB's single signon service. You should be able to find my account via my real life name.

jamesbroadhead commented:
Hi @makuhama - 

Apologies for the delay in reviewing this one. 

Could you please sign our CLA, so that we can merge it? 

https://www.mongodb.com/legal/contributor-agreement

all the best, 

James

makuhama commented:
@jamesbroadhead CLA signed

jamesbroadhead commented:
@makuhama  thank you!

I've asked @rodrigovalin to do a review pass over the PR

makuhama commented:
merged from upstream/master
fixed conflicts
removed service account mongodb-enterprise-database-pods as referrenced nowhere (not even in https://github.com/mongodb/mongodb-kubernetes-operator)

antonlisovenko commented on helm_chart/templates/database-roles.yaml:
```diff
@@ -9,17 +9,6 @@ metadata:
   namespace: {{ .Values.namespace }}
   {{- end }}
 
----
```
This service account is necessary for database pods created by the Operator

antonlisovenko commented on mongodb-enterprise-openshift.yaml:
```diff
@@ -54,39 +53,37 @@ rules:
 # definitions. The validating webhooks are optional so this can be removed if
 # necessary.
 ---
-kind: ClusterRoleBinding
```
what's the reason for this change?

antonlisovenko commented on mongodb-enterprise-openshift.yaml:
```diff
@@ -106,20 +103,19 @@ rules:
 
+
 ---
-kind: ClusterRoleBinding
```
what's the reason for this change? CSR is a cluster resource, my understanding is that to grant permissions to it we need `ClusterRoleBinding`?

antonlisovenko commented on mongodb-enterprise-openshift.yaml:
```diff
@@ -152,20 +142,41 @@ rules:
+
+
+---
+# Source: mongodb-enterprise-operator/templates/serviceaccount.yaml
```
Good idea with `imagePullSecrets` for service accounts
though the `mongodb-enterprise-openshift.yaml` already contains instructions for these two service accounts

rodrigovalin commented:
@makuhama There are a few unresolved comments that have been there for a few months now.

I'm going to close this PR. Please re-open if you think you'll have the time to address those.

Thanks again for contributing to the Operator.
